scripts/spack/packages/axom/package.py

```diff
@@ -610,3 +617,44 @@ def patch(self):
                 'PROPERTIES LINKER_LANGUAGE CXX \n LINK_FLAGS "-fopenmp"',
                 "src/axom/quest/examples/CMakeLists.txt",
             )
+
+    @run_after("build")
+    @on_package_attributes(run_tests=True)
+    def build_test(self):
+        with working_dir(self.build_directory):
+            print("Running Axom Unit Tests...")
+            make("test")
+
+    @run_after("install")
+    @on_package_attributes(run_tests=True)
+    def check_install(self):
+        """
+        Checks the spack install of axom using axom's
+        using-with-cmake example
+        """
+
+        print("Checking Axom installation...")
+        spec = self.spec
+        install_prefix = spec.prefix
+        example_src_dir = join_path(install_prefix, "examples", "axom", "using-with-cmake")
+        example_build_dir = join_path(example_src_dir, "build")
+        print("Checking using-with-cmake example...")
+        with working_dir(example_build_dir, create=True):
+            cmake_args = ["-C ../host-config.cmake", example_src_dir]
+            cmake(*cmake_args)
+            make()
+            example = Executable("./example")
+            example()
+        print("Checking using-with-make example...")
+        example_src_dir = join_path(install_prefix, "examples", "axom", "using-with-make")
```

Comment:
Should this be `using-with-blt`  instead of `using-with-make`?  

I'm not sure if `using-with-make` works when axom has dependencies like raja or mfem, unless spack adds the appropriate includes/libs/flags/defines on the command line.

See the comment at the end of: https://github.com/LLNL/axom/blob/acf7e80a12ba54b5140801a9e099a4b39fdd0f01/src/examples/using-with-make/Makefile.in#L9-L25

Reply:
This change was from @balos1 in the Spack repo (I asked him to add testing capabilties to our Spack recipe). I agree the make example is probably not ideal and the fact that it passed tests was mildly shocking to me.  We can iterate on this in another PR but I would prefer to ignore it in this PR.

Reply:
I honestly would prefer to just remove the `using-with-make` example.